File: swift/Sources/Notes/ReminderNote.swift
import Foundation

final class ReminderNote: Note<ReminderNoteModel>, AnyNote {
    override func drawCanvas() -> String {
        let m = storage.value(by: id) ?? data
        let status = m.isDone ? "выполнено" : "в процессе выполнения"
        return """
        [Напомиание]
        • id: \(id)
        • текст: \(m.text)
        • статус: \(status)
        • создано: \(dateFormatter.format(m.createdAt))
        • изменено: \(dateFormatter.format(m.updatedAt))
        """
    }

    func toggle() {
        var m = storage.value(by: id) ?? data
        m.isDone.toggle()
        m.updatedAt = Date()
        update(m)
    }

    func updateText(_ newText: String) {
        var m = storage.value(by: id) ?? data
        m.text = newText
        m.updatedAt = Date()
        update(m)
    }
}

File: kotlin/src/main/kotlin/ReminderNote.kt
import java.time.Instant

class ReminderNote(
    data: ReminderNoteModel,
    storage: Storage<ReminderNoteModel>,
    dateFormatter: DateFormatting
) : Note<ReminderNoteModel>(data, storage, dateFormatter), AnyNote {

    override fun drawCanvas(): String {
        val m = storage.valueById(id) ?: data
        val status = if (m.isDone) "выполнено" else "в процессе выполнения"
        return """
        [Напоминание]
        • id: $id
        • текст: ${m.text}
        • статус: $status
        • создано: ${dateFormatter.format(m.createdAt)}
        • изменено: ${dateFormatter.format(m.updatedAt)}
        """.trimIndent()
    }

    fun toggle() {
        val current = storage.valueById(id) ?: data
        val m = current.copy(isDone = !current.isDone, updatedAt = Instant.now())
        update(m)
    }

    fun updateText(newText: String) {
        val m = (storage.valueById(id) ?: data).copy(text = newText, updatedAt = Instant.now())
        update(m)
    }
}
